posthog/models/plugin.py

```diff
@@ -400,6 +400,14 @@ class Status(models.TextChoices):
     __repr__ = sane_repr("plugin_id", "filename", "status")
 
 
+# Some plugins are private, only certain organizations should be able to access them
+# Sometimes we want to deprecate plugins, where the first step is limiting access to organizations using them
+# Sometimes we want to test out new plugins by only enabling them for certain organizations at first
+class PluginPrivateAccess(models.Model):
+    plugin: models.ForeignKey = models.ForeignKey("Plugin", on_delete=models.CASCADE)
+    organization: models.ForeignKey = models.ForeignKey("Organization", on_delete=models.CASCADE)
```

Comment:
Django has [ManyToManyField](https://docs.djangoproject.com/en/5.0/topics/db/examples/many_to_many/) for this, it could be a field on the `Plugin` model.

I don't have a strong opinion, just pointing it out. :)